pkg/util/http.go

```diff
@@ -0,0 +1,97 @@
+package util
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/buildversion"
+)
+
+var DefaultHTTPClient *retryablehttp.Client
+
+func init() {
+	DefaultHTTPClient = retryablehttp.NewClient()
+	DefaultHTTPClient.ErrorHandler = errorHandler
+}
+
+// NewRequest returns a http.Request object with kots defaults set, including a User-Agent header.
+func NewRequest(method string, url string, body io.Reader) (*http.Request, error) {
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call newrequest: %w", err)
+	}
+
+	injectUserAgentHeader(req.Header)
+	return req, nil
+}
+
+// NewRetryableRequest returns a retryablehttp.Request object with kots defaults set, including a User-Agent header.
+func NewRetryableRequest(method string, url string, body io.Reader) (*retryablehttp.Request, error) {
+	req, err := retryablehttp.NewRequest(method, url, body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call newrequest: %w", err)
+	}
+
+	injectUserAgentHeader(req.Header)
+	return req, nil
+}
+
+func injectUserAgentHeader(header http.Header) {
+	header.Add("User-Agent", buildversion.GetUserAgent())
+}
+
+// errorHandler includes the body in the response when there is an error
```

Comment:
would this cause OOM if the body was a non-complete binary download? should probably have a cap on the body size that can be read

Reply:
good call. i made this much more simple